pkg/pillar/cmd/nodeagent/handletimers.go

```diff
@@ -197,6 +200,11 @@ func updateZedagentCloudConnectStatus(ctxPtr *nodeagentContext,
 	case types.ConfigGetSuccess:
 		log.Functionf("Config get from controller, is successful")
 		ctxPtr.lastControllerReachableTime = ctxPtr.timeTickCount
+		if !ctxPtr.configGetSuccess && ctxPtr.vaultTestStartTime > 0 {
```

Comment:
More logical thing would be to check vaultmgrreported instead of vaultTestStartTime.

Reply:
Sure. Changed